src/internal/collection/types.go

```diff
@@ -89,6 +89,9 @@ type PostgresReadWriteCollection interface {
 	ReadWriteCollection
 
 	DeleteByIndex(index *Index, indexVal string) error
+	// GetByIndex can have a large impact on database contention if used to retrieve
+	// a large number of rows. Consider using a read-only collection if possible
+	GetByIndex(index *Index, indexVal string, val proto.Message, opts *Options, f func(string) error) error
```

Comment:
I'm also open to renaming this `ExpensiveGetByIndex` or something just to make it harder to blindly switch away from read only

Reply:
I was leaning more towards renaming all uses to `ListByIndex`, but that's kind of annoying to do, so I might take care of it myself after this merges.